File: backend/models/post.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime
from enum import Enum

class PostType(str, Enum):
    data_driven = "data_driven"
    storytelling = "storytelling"
    hot_take = "hot_take"

class PostStatus(str, Enum):
    draft = "draft"
    approved = "approved"
    posted = "posted"

class PostBase(BaseModel):
    project_id: str
    type: PostType
    content: str
    status: PostStatus = PostStatus.draft
    user_edited: bool = False

class PostInDB(PostBase):
    id: str
    created_at: datetime
    
class PostUpdate(BaseModel):
    content: Optional[str] = None
    status: Optional[PostStatus] = None
    user_edited: Optional[bool] = None

File: frontend/src/types/post.ts
export type PostType = 'data_driven' | 'storytelling' | 'hot_take';
export type PostStatus = 'draft' | 'approved' | 'posted';
export type Platform = 'telegram' | 'instagram' | 'threads' | 'medium' | 'linkedin' | 'reddit';
export type ContentLanguage = 'en' | 'ru' | 'es' | 'fr' | 'de' | 'it' | 'pt' | 'ja' | 'zh' | 'ko' | 'ar' | 'hi';
export type ContentStyle = 'professional' | 'casual' | 'humorous' | 'academic';

export interface Post {
  id: string;
  project_id: string;
  type: PostType;
  platform: Platform;
  language: ContentLanguage;
  style: ContentStyle;
  content: string;
  status: PostStatus;
  user_edited: boolean;
  created_at: string;
}
